virtual_ecosystem/core/core_components.py

```diff
@@ -160,23 +160,28 @@ def __post_init__(self, config: Config) -> None:
         )
 
 
-@dataclass
 class LayerStructure:
-    """Simulation vertical layer structure.
-
-    This class defines the structure of the vertical dimension of the Virtual Ecosystem
-    from a model configuration. Four values from the ``core.layers`` configuration
-    section are used to define a set of vertical layers and their heights (or relative
-    heights): ``canopy_layers``, ``soil_layers``, ``above_canopy_height_offset`` and
-    ``surface_layer_height``. These values are validated and then assigned to attributes
-    of this class. The ``n_layers`` and ``layer_roles`` attributes report the total
-    number of layers in the vertical dimension and an array giving the vertical sequence
-    of layer roles.
-
-    The layer structure is shown below, along with values from the default
-    configuration. All heights are in metres relative to ground level and the canopy
-    layer heights are defined dynamically by the
-    :class:`~virtual_ecosystem.models.plants.plants_model.PlantsModel`.
+    """Vertical layer structure of the Virtual Ecosystem.
+
+    This class defines the structure of the vertical dimension of a simulation using the
+    Virtual Ecosystem. The vertical dimension is divided into a series of layers,
+    ordered from above the canopy to the bottom of the soil, that perform different
+    roles in the simulation. The layers are defined using the following five
+    configuration settings from the ``[core.layers]`` section.
+
+    * ``above_canopy_height_offset``: the height above the canopy top of the first layer
+      role ``above``, which is used as the measurement height of reference climate data.
+    * ``canopy_layers``: a fixed number of layers with the ``canopy`` role. Not all of
+      these necessarily contain canopy during a simulation as the canopy structure
+      within these layers is dynamic.
+    * ``surface_layer_heights``: the height above ground level of the ground surface
+      atmospheric layer.
+    * ``soil_layers``: the number and depths of soil layers.
```

Comment:
Should we add a comment that this is different from the layer thickness?

Reply:
I've updated the description throughout the docstring to make it clear that ``soil_layers`` are the depths of horizons relative to the surface and thicknesses are individual layer... thicknesses!